### DISCUSSION TITLE
Structured errors

### DISCUSSION THREAD

**Discussion by mholt** [wailsapp/wails#2048] [Ideas]
If a bound function returns an error, I believe  currently the error consists only of an error string. At least, that's what I've seen so far.

I wonder if it would be useful to make errors a struct (kind of like those with `fetch` calls, where you get all sorts of info about an error like the response status, headers, etc) -- containing the error string of course, but perhaps also the name of the error type, and perhaps common known errors, like exported errors in the Go std lib, since the frontend can't type-assert errors AFAIK. (For example, `errors.Is(err, fs.ErrNotExist)`, etc, and similar errors in `os` or `net`).

For now, as a workaround I'm just manually detecting the errors I'm interested in (in my case, `fs.ErrNotExist`) and adding a keyword to the error string (`fmt.Errorf("%w (ErrNotExist)"`) then I can use `error.includes("ErrNotExist")` on the front-end to detect that kind of error.

Just wondering what everyone's thoughts are on something like this. :smiley:  Depending on availability and difficulty I could potentially submit a patch!

**Comment by leaanthony** [comment]
I endorse this message!!
